src/vsphere_cpi/lib/cloud/vsphere/nsxt_provider.rb

```diff
@@ -197,6 +220,51 @@ def set_vif_type(vm, vm_type_nsxt)
       end
     end
 
+    # Add vm to given list of static Load Balancer Server Pools
+    # @param [Resources::VM] vm
+    # @param [[NSXT::LbPool, integer][]] load_balancer_pools Array of Load_balancer_pool and port no
+    def add_vm_to_server_pools(vm, load_balancer_pools)
+      return if load_balancer_pools.nil? ||  load_balancer_pools.empty?
+      Bosh::Retryable.new(
+        tries: 50,
+        sleep: ->(try_count, retry_exception) { 2 },
+        on: [VirtualMachineIpNotFound]
+      ).retryer do |i|
+        vm_ip =  vm.mob.guest&.ip_address
+        raise VirtualMachineIpNotFound.new(vm) unless vm_ip
+        load_balancer_pools.each do |load_balancer_pool, port_no|
+          @logger.info("Adding vm: '#{vm.cid}' with ip:#{vm_ip} to ServerPool: #{load_balancer_pool} on Port: #{port_no} ")
+          pool_member = NSXT::PoolMemberSetting.new(ip_address: vm_ip, port: port_no)
+          pool_member_setting_list = NSXT::PoolMemberSettingList.new(members: [pool_member])
+          services_svc.perform_pool_member_action(load_balancer_pool.id, pool_member_setting_list, 'ADD_MEMBERS')
+        end
+      end
+    end
+
+    # Returns an array of Static Load Balancer Server Pools and Dynamic Load Balancer Pools
+    # For Static Server Pools corresponding port no is also returned
+    # @param [array] server_pools It is an array of hashes with server_pool names and port
+    def retrieve_server_pools(server_pools)
+      return [] if server_pools.nil? || server_pools.empty?
+      server_pools_by_name = services_svc.list_load_balancer_pools.results.each_with_object({}) do |server_pool, hash|
+        hash[server_pool.display_name] = server_pool
+      end
+
+      missing = server_pools.reject do |server_pool|
+        server_pools_by_name.key?(server_pool['name'])
+      end
+      raise ServerPoolsNotFound.new(*missing) unless missing.empty?
+
+      static_server_pools, dynamic_server_pools = [],[]
```

Comment:
Use the `Enumerable#partition` method for this logic.

Reply:
Cannot as I have updated the logic and actually not splitting array into 2 as both arrays have different elements. Static pools also have ports being returned with them.